Makefile

```diff
@@ -0,0 +1,89 @@
+
+# NOTE: This is not the most perfect makefile.
+# There's a few things we could do to improve performance.
+# But since this is new, we can strive for clarity and simplicity here.
+# Could use many of these suggestions:
+# 	https://docs.cloudposse.com/reference/best-practices/make-best-practices/
+
+
+# On _some_ systems, make might not default to bash, which might produce
+# unexpected output/errors. It's assumed that all of the recipes here are written
+# in bash syntax.
+SHELL = /bin/bash
+.SHELLFLAGS = -e -u -o pipefail -c
```

Comment:
These are some of my favorite shell flags.. These make it less likely that we make silly errors in the bash code below. Highly recommend reading up on these flags: https://tldp.org/LDP/abs/html/options.html.